### PR TITLE
[SYCLomatic] adding make_transform_output_iterator tests

### PR DESCRIPTION
**Pull request by danhoeflinger**
Adding tests for dpct::make_transform_output_iterator.  

Meant to test code added in https://github.com/oneapi-src/SYCLomatic/pull/361.  Tests will not pass until this PR is merged.

### CODE REVIEW

File: help_function/src/onedpl_test_transform_output_iterator.cpp [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+// ===----------------------------------------------------------------------===//
+
+#include "oneapi/dpl/algorithm"
+#include "oneapi/dpl/execution"
+#include "oneapi/dpl/iterator"
+
+#include "dpct/dpct.hpp"
+#include "dpct/dpl_utils.hpp"
+
+#include <CL/sycl.hpp>
```
Comment (mmichel11) [inline help_function/src/onedpl_test_transform_output_iterator.cpp]:
```suggestion
#include <sycl/sycl.hpp>
```

File: help_function/help_function.xml [hunk 2]
```diff
@@ -121,6 +121,7 @@
     <test testName="onedpl_test_stable_partition" configFile="config/TEMPLATE_help_function_skip_cuda_backend.xml" />
     <test testName="onedpl_test_tabulate" configFile="config/TEMPLATE_help_function_skip_cuda_backend.xml" />
     <test testName="onedpl_test_transform_if" configFile="config/TEMPLATE_help_function_skip_cuda_backend.xml" />
+    <test testName="onedpl_test_transform_output_iterator" configFile="config/TEMPLATE_help_function_skip_cuda_backend.xml" />
```
Comment (mmichel11) [inline help_function/help_function.xml:130]: This test internally uses `sycl::malloc_shared`. Is it safe to use the `config/TEMPLATE_help_function_skip_cuda_backend.xml/` or should we use `config/TEMPLATE_help_function_usmnone.xml`?

File: help_function/src/onedpl_test_transform_output_iterator.cpp [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+    return failed_tests;
+  }
+};
+
+template <typename OutputIterator1, typename UnaryFunc>
+auto attempt_to_dangle(OutputIterator1 out1, UnaryFunc unary) {
+  auto toi = oneapi::dpl::make_transform_output_iterator(out1, unary);
+  // Leave scope of the transform output iterator, then assign data
+  // to the wrapper after the fact.
+  return *toi;
```
Comment (mmichel11) [inline help_function/src/onedpl_test_transform_output_iterator.cpp]: Just as clarification, do we intend to the behavior to be that the transformation is still performed? I see the expected result is filled with `init * init`.

File: help_function/src/onedpl_test_transform_output_iterator.cpp [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+#include "oneapi/dpl/algorithm"
+#include "oneapi/dpl/execution"
+#include "oneapi/dpl/iterator"
+
+#include "dpct/dpct.hpp"
+#include "dpct/dpl_utils.hpp"
+
+#include <CL/sycl.hpp>
+
+#include <iomanip>
```
Comment (mmichel11) [inline help_function/src/onedpl_test_transform_output_iterator.cpp]: Maybe I missed it, but is the `iomanip` header used anywhere?

**Review by mmichel11 — CHANGES_REQUESTED** [review]

**Review by tomflinda — COMMENTED** [review]
pls re-run the CI test.

File: help_function/src/onedpl_test_transform_output_iterator.cpp (hunk 4 above)
Comment (danhoeflinger) [reply help_function/src/onedpl_test_transform_output_iterator.cpp]: nope, removed.

File: help_function/src/onedpl_test_transform_output_iterator.cpp (hunk 3 above)
Comment (danhoeflinger) [reply help_function/src/onedpl_test_transform_output_iterator.cpp]: Yes, these tests are basically just insuring that there is no dependence on the transform output iterator existing for the individual dereferenced elements to still perform the transformation.  It is working as intended, performing the transformation.  
However, I've removed the one specific to zip iterators as I think it is redundant and excessive.

File: help_function/help_function.xml (hunk 2 above)
Comment (danhoeflinger) [reply help_function/help_function.xml:130]: I've rewritten the tests to not rely upon USM memory.

File: help_function/src/onedpl_test_transform_output_iterator.cpp (hunk 1 above)
Comment (danhoeflinger) [reply help_function/src/onedpl_test_transform_output_iterator.cpp]: done, Thanks.

**Review by mmichel11 — APPROVED** [review]
LGTM

**Comment by danhoeflinger** [comment]
@tomflinda  I believe this PR should be ready to merge, the failing CI is unrelated.  I do not have privileges to merge.

**Review by tomflinda — APPROVED** [review]
LGTM

**Pull request merged**